src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -28,6 +29,19 @@
 
 ;; FUNCTIONS
 
+(defn clj->pgobject
```

Comment:
Maybe two extra helpers here would be nice that do the work of
```
(if (scf-utils/postgres?)
  scf-utils/pgobject->clj
  json/parse-string)
```
and
```
(if (scf-utils/postgres?)
  (scf-utils/clj->pgobject (:metrics report))
  (json/generate-string (:metrics report)))
```
So that the pieces of code needing to know about postgres or not are minimized.